src/__tests__/sourceNodes.js

```diff
@@ -75,4 +76,151 @@ describe('sourceNodes', () => {
     expect(createNodeId).not.toBeCalled();
     expect(reporter.panic.mock.calls.length).toBe(1);
   });
+
+  test('sourceNodes with list and readAll, success response', async () => {
+    const options = {
+      apiKey: 'key',
+      serviceId: 'id',
+      endpoint: 'point',
+      format: 'list',
+      readAll: true,
+      version: 'v1',
+    };
+    mockResponse = {
+      statusCode: 200,
+      body: { contents: [{ id: '1' }, { id: '2' }], totalCount: 2 },
+    };
+    await sourceNodes({ actions, createNodeId, reporter }, options);
+    expect(actions.createNode.mock.calls.length).toBe(2);
+    expect(createNodeId.mock.calls.length).toBe(2);
+    expect(reporter.panic).not.toBeCalled();
+  });
+
+  test('sourceNodes with list and readAll, success response', async () => {
+    const options = {
+      apiKey: 'key',
+      serviceId: 'id',
+      endpoint: 'point',
+      format: 'list',
+      readAll: true,
+      version: 'v1',
+      query: {
+        limit: 2,
+      },
+    };
+    mockResponse = {
+      statusCode: 200,
+      body: {
+        contents: [{ id: '1' }, { id: '2' }, { id: '3' }],
+        totalCount: 2,
+      },
+    };
+    await sourceNodes({ actions, createNodeId, reporter }, options);
+    expect(actions.createNode.mock.calls.length).toBe(3);
+    expect(createNodeId.mock.calls.length).toBe(3);
+    expect(reporter.panic).not.toBeCalled();
+  });
```

Comment:
When limit is 2, it is unnatural that mockResponse.body.contents has 3 lengths, so I'd like you to correct it so that the consistency matches.

ex)
```
limit: 3
mockResponse = {
  statusCode: 200,
  body: {
    contents: [{ id: '1' }, { id: '2' }, { id: '3' }],
    totalCount: 12,
  },
}
expect(actions.createNode.mock.calls.length).toBe(12);
expect(createNodeId.mock.calls.length).toBe(12);
```